src/libraries/System.Runtime.Numerics/src/System/Number.BigInteger.cs

```diff
@@ -1437,4 +1309,97 @@ internal static bool TryFormatBigInteger(BigInteger value, ReadOnlySpan<char> fo
             }
         }
     }
+
+    internal interface IBigIntegerHexOrBinaryParsingInfo<TParsingInfo, TChar>
+        where TParsingInfo : struct, IBigIntegerHexOrBinaryParsingInfo<TParsingInfo, TChar>
+        where TChar : unmanaged, IBinaryInteger<TChar>
+    {
+        static abstract int BitsPerDigit { get; }
+
+        static virtual int DigitsPerBlock => sizeof(uint) * 8 / TParsingInfo.BitsPerDigit;
+
+        static abstract NumberStyles BlockNumberStyle { get; }
+
+        static abstract uint GetSignBitsIfValid(uint ch);
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        static virtual bool TryParseUnalignedBlock(ReadOnlySpan<TChar> input, out uint result)
+        {
+            if (typeof(TChar) == typeof(char))
+            {
+                return uint.TryParse(MemoryMarshal.Cast<TChar, char>(input), TParsingInfo.BlockNumberStyle, null, out result);
+            }
+
+            throw new NotSupportedException();
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        static virtual bool TryParseSingleBlock(ReadOnlySpan<TChar> input, out uint result)
+            => TParsingInfo.TryParseUnalignedBlock(input, out result);
```

Comment:
This leaves the space for vectorized `Vector128<char>` -> `uint` conversion. This may or may not be necessary if such optimization is done in uint side.